rules/manager.go

```diff
@@ -37,9 +37,9 @@ import (
 const (
 	namespace = "prometheus"
 
-	ruleTypeLabel     = "rule_type"
-	alertingRuleType  = "alerting"
-	recordingRuleType = "recording"
+	RuleTypeLabel     = "rule_type"
```

Comment:
In general I agree with the only-export-what-you-use approach. The only unwanted consequence I've seen quite often by now is, that when people actually need to make something exported, they just go into the code and brute-force something into it without putting thought into the thing they are exposing.

The result is stuff like `BaseLabelsWithJobAndInstanceLabel()` in an interface... which is even redundant as there's already `BaseLabels()`.

It would be great if we could keep an extra eye on things becoming exported so that they are exported in a generic and meaningful way and not tailored to a single use case.

So in this case I'd also go ahead and turn `Eval` into `eval`.
Do we agree that it's a bit weird to export `AlertingRule` but not `recordingRule`?